src/btree/bt_handle.c

```diff
@@ -768,9 +824,15 @@ int
 __wt_btree_new_leaf_page(WT_SESSION_IMPL *session, WT_REF *ref)
 {
     WT_BTREE *btree;
+    uint8_t previous_state;
 
+    previous_state = ref->state;
```

Comment:
I believe it's too late to save the previous state here - it has already been shifted to locked by the caller.

Reply:
Good point!